python/clx/analytics/dga_detector.py

```diff
@@ -63,26 +63,25 @@ def save_model(self, file_path):
         }
         super()._save_model(checkpoint, file_path)
 
-    def train_model(self, detector_dataset):
+    def train_model(self, dataloader):
```

Comment:
what about hiding the use of dataloader and dataset? might be easier for someone to just pass cudf dataframe directly along with batch_size and avoid the preprocessing. we can do that under the hood.

Reply:
@efajardo-nv we can pass the dataframe directly, but `train_model` function is called for every epoch and preprocessing happens repeatedly. Do you think adding a separate function which reads cudf dataframe and returns dataloader is good idea?

Reply:
can we have `train_model` perform the epochs internally as well?

Reply:
I can make changes within the train model function to run multiple epochs.

Reply:
Updated with the suggested changes. But currently `train_model` test is failing with at cuML function `train_test_split`. Thank you!